cmd/migration-tool/main.go

```diff
@@ -0,0 +1,524 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	v1 "github.com/openshift/api/image/v1"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned"
+	imageclient "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	"github.com/openshift/release-controller/pkg/apis/release/v1alpha1"
+	"github.com/openshift/release-controller/pkg/client/clientset/versioned"
+	releasev1alpha1 "github.com/openshift/release-controller/pkg/client/clientset/versioned/typed/release/v1alpha1"
+	releasecontroller "github.com/openshift/release-controller/pkg/release-controller"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	// defaultAggregateProwJobName the default ProwJob to call if no override is specified
+	defaultAggregateProwJobName = "release-openshift-release-analysis-aggregator"
+)
+
+type options struct {
+	Execute       bool
+	Name          string
+	Namespace     string
+	ProwNamespace string
+	Location      string
+}
+
+func main() {
+	original := flag.CommandLine
+	klog.InitFlags(original)
+	original.Set("alsologtostderr", "true")
+	original.Set("v", "2")
+
+	opt := &options{
+		Namespace:     "ocp",
+		Name:          "release",
+		ProwNamespace: "ci",
+	}
+
+	cmd := &cobra.Command{
+		Run: func(cmd *cobra.Command, arguments []string) {
+			if err := opt.Run(); err != nil {
+				klog.Exitf("Run error: %v", err)
+			}
+		},
+	}
+
+	flagset := cmd.Flags()
+	flagset.BoolVar(&opt.Execute, "execute", opt.Execute, "Create ReleasePayload resources on currently configured cluster")
+	flagset.StringVar(&opt.Namespace, "namespace", opt.Namespace, "Namespace where the image stream, containing releases, resides")
+	flagset.StringVar(&opt.Name, "name", opt.Name, "Name of the image stream, containing releases, to process")
+	flagset.StringVar(&opt.ProwNamespace, "prow-namespace", opt.ProwNamespace, "The namespace where Prow jobs are executed.")
+	flagset.StringVar(&opt.Location, "location", opt.Location, "The path to a folder containing the release payloads from a previous invocation.")
+	flagset.AddGoFlag(original.Lookup("v"))
+
+	if err := cmd.Execute(); err != nil {
+		klog.Exitf("Execute error: %v", err)
+	}
+}
+
+func (o *options) Run() error {
```

Comment:
important style note, place this in a non-main for importing and integration testing capabilities.  it doesn't cost anything.